apps/api/src/app/step-schemas/dtos/step-schema.dto.ts

```diff
@@ -1,6 +1,83 @@
 import { JSONSchema } from 'json-schema-to-ts';
+import { StepType } from '@novu/shared';
+
+import { ActionStepEnum, ChannelStepEnum } from '@novu/framework/internal';
+
+const chatProperties = {
+  body: { type: 'chatBody' },
+};
+
+const emailProperties = {
+  subject: { type: 'emailSubject' },
+  body: { type: 'emailBody' },
+};
+
+const inAppProperties = {
+  subject: { type: 'inAppSubject' },
+  body: { type: 'inAppBody' },
+  avatar: { type: 'inAppAvatar' },
+  primaryAction: { type: 'inAppPrimaryAction' },
+  secondaryAction: { type: 'inAppSecondaryAction' },
+  data: { type: 'inAppData' },
+  redirect: { type: 'inAppRedirect' },
+};
+
+const pushProperties = {
+  subject: { type: 'pushSubject' },
+  body: { type: 'pushBody' },
```

Comment:
why not just text? it's going to make the FE work super problematic

Reply:
as discussed in the meeting, in this solution, the `client knows something` rather than `nothing` the second solution suggested in the DX guide. Here, the client needs to know how to render `pushBody`. In the future, we can expand this approach to provide even more data to the client if needed.